core/main/src/firebolt/handlers/internal_rpc.rs

```diff
@@ -145,3 +182,22 @@ impl RippleRPCProvider<InternalImpl> for InternalProvider {
         (InternalImpl { state }).into_rpc()
     }
 }
+
+impl PolicyState {
+    pub async fn add_policy_identifier_alias(&mut self, policy: PolicyIdentifierAlias) {
+        let mut policy_identifiers_alias = self.policy_identifiers_alias.write().unwrap();
+        if policy.policy_identifier_alias.is_empty() {
+            policy_identifiers_alias.clear();
+            return;
+        }
+        for identifier in policy.policy_identifier_alias {
+            if !policy_identifiers_alias.contains(&identifier) {
```

Comment:
I'm not sure if this is the correct logic. The entire vector has to be replaced.